src/decode.rs

```diff
@@ -19,47 +19,36 @@ use super::error::{Error, Result};
 #[derive(Clone, Debug, PartialEq)]
 // TODO can we make this require less copying?
 pub struct Event {
+    // TODO(mmk) If this is empty at dispatch time, we need to set this to message
```

Comment:
The todos are notes to myself at this point to remember to address stuff. They get addressed later.

Having Event contain a field map is silly since we only care about 2 types of data. We make that explicit in this commit. NOTE: This change will require some updates in the SDK but they are small.